doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+|:---:|:----------:|:--------------:|:----------------|
+| 0.1 | 18/05/2022 | Lior Avramov   | Initial version |
+
+## Scope
+This document describes the high level design of IP interface loopback action feature in SONiC
+
+## List of figures
+[Figure 1: feature overview](#figure-1-feature-overview)  
+[Figure 2: interface vlan](#figure-2-interface-vlan)  
+
+# 1 Introduction
+## 1.1 Feature overview
+IP interface loopback action is a feature that allows user to change the way router handles routed packets for which egress port equals to ingress port.
+1. When loopback action is configured to drop, those packets will be dropped.
+2. When loopback action is configured to forward, those packets will be forwarded.
+
+![feature overview](images/feature_overview.svg "Figure 1: feature overview")
+###### Figure 1: feature overview
+
+## 1.2 Requirements
+### 1.2.1 Functionality
+**This feature will support the following functionality:**
+1. Loopback action can be set on IP interface. 
+2. IP Interface can be one of the following:
+    1. Interface vlan
+    2. Interface Ethernet
+    3. Interface port-channel 
+3. Loopback action can be set to "drop" or "forward". 
+4. CLI does not allow to set loopback action on non IP interface. Interface is considered as IP interface if it has entry in:
+    1. VLAN_INTERFACE table for interface vlan (e.g. VLAN_INTERFACE|Vlan100)
+    2. INTERFACE table for interface Ethernet (e.g. INTERFACE|Ethernet232)
+    3. PORTCHANNEL_INTERFACE table for interface port-channel (e.g. PORTCHANNEL_INTERFACE|PortChannel1)    
+5. The configured loopback action can be viewed by show command in CLI.
+
+### 1.2.2 Command interface
+**This feature will support the following commands:**
+1. config: set IP interface loopback action.
+2. show: display the configured loopback action.
+
+### 1.2.3 Error handling
+**This feature will provide error handling for the next situations:**
+1. In CLI: invalid action, invalid interace, non ip interface
+2. In SWSS: invalid action, invalid interace, non ip interface, invalid DB operation
+
+### 1.2.4 Event logging
+**This feature will provide event logging for:** loopback action set on IP interface.
+
+###### Table 1: Event logging
+| Event                                     | Severity |
+|:------------------------------------------|:---------|
+| IP interface loopback action set: success | NOTICE   |
+| IP interface loopback action set: error   | ERROR    |
+
+# 2 Design
+## 2.1 Overview
+Intfmgrd subscribes for changes in L3 interface tables VLAN_INTERFACE, INTERFACE and PORTCHANNEL_INTERFACE in config DB. Whenever there is a change in one of these tables, intfmgrd is notified. Intfmgrd then updates INTF_TABLE table in app DB. Upon change in INTF_TABLE, intforch is notified.
+Intforch performs validity checks, call the relevant SAI API and updates internal cache. There is an existing mechanizm in intforch that postpones handling of changes in INTF_TABLE until interface is ready, we will use it.
```

Comment:
mechanizm is a polish word :-) use mechanism 

i would suggest to have it in a different way.
Based on exition sw arch, this ... will be based on the current interface configuration flows in interface manager as well as interface orch.
if there is a flow you want to show, diagram it or have it like bullets.

Reply:
Done.